packages/block-editor/src/components/media-replace-flow/index.js

```diff
@@ -0,0 +1,224 @@
+		<FormFileUpload
+			onChange={ uploadFiles }
+			accept={ allowedTypes }
+			multiple={ multiple }
+			render={ ( { openFileDialog } ) => {
+				return (
+					<MenuItem
+						icon="upload"
+						onClick={ () => {
+							openFileDialog();
+						} }
+					>
+						{ __( 'Upload' ) }
+					</MenuItem>
+				);
+			} }
+		/>
+	);
+
+	const URLButton = (
+		<MenuItem
+			icon="admin-links"
+			onClick={ () => ( setShowURLInput( ! showURLInput ) ) }
+			aria-expanded={ showURLInput }
+		>
+			<div> { __( 'Insert from URL' ) } </div>
+		</MenuItem>
+	);
+
+	let urlInputUIContent;
+	if ( showEditURLInput ) {
+		urlInputUIContent = ( <LinkEditor
+			onKeyDown={ stopPropagationRelevantKeys }
+			onKeyPress={ stopPropagation }
+			value={ mediaURLValue }
+			isFullWidthInput={ true }
+			hasInputBorder={ true }
+			onChangeInputValue={ ( url ) => ( setMediaURLValue( url ) ) }
+			onSubmit={ ( event ) => {
+				event.preventDefault();
+				onSelectURL( mediaURLValue );
+				setShowEditURLInput( ! showEditURLInput );
+				editMediaButtonRef.current.focus();
+			} }
+		/> );
+	} else {
+		urlInputUIContent = ( <LinkViewer
+			isFullWidth={ true }
+			className="editor-format-toolbar__link-container-content block-editor-format-toolbar__link-container-content"
+			url={ mediaURLValue }
+			onEditLinkClick={ () => ( setShowEditURLInput( ! showEditURLInput ) ) }
+		/> );
+	}
+
+	const urlInputUI = ( showURLInput && <div className="block-editor-media-flow__url-input">
+		{ urlInputUIContent }
+	</div> );
+
+	const onClose = () => {
+		editMediaButtonRef.current.focus();
+	};
+
+	const onClickOutside = () => ( setShowMediaReplaceOptions( false ) );
+
+	const editMediaButton = (
+		<MediaUploadCheck>
+			<MediaUpload
+				onSelect={ ( media ) => selectMedia( media ) }
+				onClose={ () => setShowMediaReplaceOptions( true ) }
+				allowedTypes={ allowedTypes }
+				render={ ( { open } ) => (
+					<>
+						<Toolbar className={ 'mediaflow-replace components-dropdown-menu' }>
+							<Button
+								ref={ editMediaButtonRef }
+								className={ 'components-icon-button components-dropdown-menu__toggle' }
+								onClick={ () => {
+									setShowMediaReplaceOptions( ! showMediaReplaceOptions );
+								} }
+							>
+								<span className="components-dropdown-menu__label" > { name } </span>
+								<span className="components-dropdown-menu__indicator" />
+							</Button>
+							{ showMediaReplaceOptions &&
+								<Popover
+									onClickOutside={ onClickOutside }
+									onClose={ onClose }
+								>
+									<>
+										<NavigableMenu>
+											<MenuItem
+												icon="admin-media"
+												onClick={ open }
+											>
+												{ __( 'Open Media Library' ) }
+											</MenuItem>
+											{ fileUploadButton }
+											{ URLButton }
+										</NavigableMenu>
+										{ urlInputUI }
+									</>
+								</Popover>
+							}
+						</Toolbar>
+					</>
+				) }
+			/>
+		</MediaUploadCheck>
+	);
+
+	return (
+		<>
```

Comment:
No need for the fragment, this can just be `return editMediaButton`. Having said that, the whole of editMediaButton could just be inlined.